ci/cirrus_general_ci.yml

```diff
@@ -3,6 +3,21 @@
 # cibuildwheel also runs tests during the wheel build process, so there's no need
 # to have duplication.
 
+modified_clone: &MODIFIED_CLONE
+  # makes sure that for a PR the CI runs against a merged main
+  clone_script: |
+    if [ -z "$CIRRUS_PR" ]; then
+      # if you're not in a PR then clone against the branch name that was pushed to.
+      git clone --recursive --branch=$CIRRUS_BRANCH https://x-access-token:${CIRRUS_REPO_CLONE_TOKEN}@github.com/${CIRRUS_REPO_FULL_NAME}.git $CIRRUS_WORKING_DIR
+      git reset --hard $CIRRUS_CHANGE_IN_REPO
+    else
+      # it's a PR so clone the main branch then merge the changes from the PR
+      git clone --recursive https://x-access-token:${CIRRUS_REPO_CLONE_TOKEN}@github.com/${CIRRUS_REPO_FULL_NAME}.git $CIRRUS_WORKING_DIR
```

Comment:
@rgommers, is this what we would like for the git clone? i.e. clone `scipy/scipy` (`CIRRUS_REPO_FULL_NAME`), then fetch the changes for a given PR, then merge those changes into main?

This won't work if the changes aren't mergeable into main.

Reply:
Or is something like this preferred:
```
git clone --recursive https://x-access-token:${CIRRUS_REPO_CLONE_TOKEN}@github.com/${CIRRUS_REPO_FULL_NAME}.git $CIRRUS_WORKING_DIR
git fetch origin pull/$CIRRUS_PR/head:pull/$CIRRUS_PR
git checkout pull/$CIRRUS_PR
git pull --no-commit origin main
```
i.e. merging main onto the PR.

Reply:
That seems fine to me either way, doesn't matter which branch is merged into which. So let's go with this, assuming you're happy with it?

Reply:
Let's stay with what I've got already. Seems to work for me locally. It will break if the PR can't be merged without conflict.